packages/monitoring/src/serviceMonitor/alivenessChecker.ts

```diff
@@ -77,3 +83,111 @@ export class ServiceMonitor {
+   *
+   * @returns {Promise<{alive: boolean, error?: Error}>} - A promise that resolves with the liveness status of the service.
+   */
+  private async pingService(service: ILivenessChecker) {
+    const status = await service.isAlive();
+    if ("disconnect" in service) {
+      await (service as IDisconnectHandler).disconnect();
+    }
+    return status;
+  }
+
+  /**
+   * Attempts to find a reachable service URL from a list of provided URLs.
+   *
+   * This method iterates through the list of URLs, repeatedly pinging the service to
+   * check if it is alive. If a reachable URL is found, it is returned. If all URLs
+   * are exhausted without finding a reachable service, an error is thrown.
+   *
+   * @param {string[]} urls - An array of service URLs to check for reachability.
+   * @param {ILivenessChecker} service - An instance of ILivenessChecker that provides methods
+   * to check the service's liveness and manage service URLs.
+   *
+   * @returns {Promise<string>} - A promise that resolves with the reachable service URL.
+   *
+   * @throws {Error} - Throws an error if no reachable service URL is found after checking all URLs.
+   *
+   */
+  async getAvailableServiceStack(urls: string[], service: ILivenessChecker) {
+    let error: Error;
+    for (let i = 0; i < urls.length; i++) {
+      if (i != 0) await service.updateUrl(urls[i]);
+      monitorEvents.log(`${service.serviceName()}: pinging ${service.serviceUrl()}`, "gray");
+      for (let retry = 0; retry < this.retries; retry++) {
+        const status = await this.pingService(service);
+        if (status.alive) {
+          monitorEvents.log(`${service.serviceName()} on ${service.serviceUrl()} Success!`, "green");
+          return service.serviceUrl();
+        }
+        error = status.error;
+      }
+      monitorEvents.log(
+        `${service.serviceName()}: failed to ping ${service.serviceUrl()} after ${this.retries} retries; ${error}`,
+        "red",
+      );
+    }
+    throw new Error(` Failed to reach ${service.serviceName()} on all provided stacks`);
+  }
+
+  /**
+   * Fetches available service URLs for tfChain and RMB services.
+   *
+   * This method checks the availability of URLs for services by utilizing
+   * the `getAvailableServiceStack` method. It updates the `result` property with the reachable URLs.
+   *
+   * @returns {Promise<ServicesUrls>} - A promise that resolves with an object containing the available service URLs.
+   */
+  async GetAvailableServices(): Promise<ServicesUrls> {
+    if (this[ServiceName.tfChain]?.length > 0)
+      this.result[ServiceName.tfChain] = await this.getAvailableServiceStack(
+        this[ServiceName.tfChain],
+        new TFChainMonitor(this[ServiceName.tfChain][0]),
+      );
+    if (this[ServiceName.RMB]?.length > 0) {
+      if (!this.result[ServiceName.tfChain]) throw new Error("Can't validate RMB urls; There is no Chain ulr provided");
```

Comment:
not sure if we should make the RMB independent form tfchain, in case i need to get available stack over predefined or provided  tfchain url